code/controllers/subsystems/ticker.dm

```diff
@@ -322,6 +330,47 @@ var/datum/controller/subsystem/ticker/SSticker
 
 	return 1
 
+/datum/controller/subsystem/ticker/proc/setup_player_ready_list()
+	// One big loop at the start once SSticker comes up to catch anyone who readied before-hand
+	// Afterward the list updates on a per-user basis in menu.dm and logout.dm
+	if(current_state >= GAME_STATE_PLAYING || !SSjobs)
+		return // how
+
+	for(var/mob/abstract/new_player/NP in player_list)
+		if(NP.ready)
+			update_ready_list(NP)
+
+	total_players_ready = LAZYLEN(ready_player_jobs)
+
+/datum/controller/subsystem/ticker/proc/update_ready_list(var/mob/abstract/new_player/NP)
+	if(current_state >= GAME_STATE_PLAYING || !SSjobs || !istype(NP))
+		return // don't bother once the game has started
+
+	if(NP.ready)
+		ready_player(NP.client.prefs)
+	else
+		unready_player(NP.last_ready_name)
+
+	// TODO: (maybe) -- sort the list by name/job? subdivide into sections? better handling than this hackjob?
```

Comment:
Sorting by department then alphabetically might be cool? Or just alphabetically.